state-sync/src/coordinator.rs

```diff
@@ -914,71 +977,89 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
+    /// * Triggers post-commit actions based on new local state (after successfully processing a chunk).
+    async fn process_chunk_response(
+        &mut self,
+        peer: &PeerNetworkId,
+        response: GetChunkResponse,
+    ) -> Result<(), Error> {
+        // Ensure consensus isn't running, otherwise we might get a race with storage writes.
         if self.is_consensus_executing() {
-            error!(LogSchema::event_log(
-                LogEntry::ProcessChunkResponse,
-                LogEvent::ConsensusIsRunning
-            )
-            .peer(peer)
-            .error(&format_err!(
-                "Received a chunk response but consensus is now running so we cannot process it!"
-            )));
-            return;
+            let error = Error::ConsensusIsExecuting;
+            error!(LogSchema::new(LogEntry::ProcessChunkResponse,)
+                .peer(peer)
+                .error(&error.clone().into()));
+            return Err(error);
         }
 
-        let new_txns = response.txn_list_with_proof.transactions.clone();
-        // Part 1: check response, validate and store chunk
-        // any errors thrown here should be for detecting actual bad chunks
-        if let Err(e) = self.apply_chunk(peer, response) {
-            // count, log, and exit
-            error!(
-                LogSchema::event_log(LogEntry::ProcessChunkResponse, LogEvent::ApplyChunkFail)
-                    .peer(peer)
-                    .error(&e)
-            );
-            counters::APPLY_CHUNK_COUNT
-                .with_label_values(&[
-                    &peer.raw_network_id().to_string(),
-                    &peer.peer_id().to_string(),
-                    counters::FAIL_LABEL,
-                ])
-                .inc();
-            return;
+        // Validate the response and store the chunk if possible.
+        // Any errors thrown here should be for detecting bad chunks.
+        match self.apply_chunk(peer, response.clone()) {
+            Ok(()) => {
+                counters::APPLY_CHUNK_COUNT
+                    .with_label_values(&[
+                        &peer.raw_network_id().to_string(),
+                        &peer.peer_id().to_string(),
+                        counters::SUCCESS_LABEL,
+                    ])
+                    .inc();
+            }
+            Err(error) => {
+                error!(LogSchema::event_log(
+                    LogEntry::ProcessChunkResponse,
+                    LogEvent::ApplyChunkFail
+                )
+                .peer(peer)
+                .error(&error.clone().into()));
+                counters::APPLY_CHUNK_COUNT
+                    .with_label_values(&[
+                        &peer.raw_network_id().to_string(),
+                        &peer.peer_id().to_string(),
+                        counters::FAIL_LABEL,
+                    ])
+                    .inc();
+                return Err(error);
+            }
```

Comment:
Another one where I'm curious why you changed from `if let` to `match`

Unless there's a good reason, it's usually best to leave it where it's at

Reply:
Here, I was grouping code by function. If apply chunk works, we update the counter. If it doesn't, we log and update the counter. Granted, this could have been achieved by using `if` and `else`, but `match` is equivalent, and in that case, it's nothing more than preference.